src/stores/SpaceStore.tsx

```diff
@@ -366,16 +366,18 @@ export class SpaceStoreClass extends AsyncStoreWithClient<IState> {
     }
 
     public getParents(roomId: string, canonicalOnly = false): Room[] {
+        const userId = this.matrixClient?.getUserId();
         const room = this.matrixClient?.getRoom(roomId);
         return room?.currentState.getStateEvents(EventType.SpaceParent)
-            .filter(ev => {
+            .map(ev => {
                 const content = ev.getContent();
-                if (!content?.via?.length) return false;
-                // TODO apply permissions check to verify that the parent mapping is valid
-                if (canonicalOnly && !content?.canonical) return false;
-                return true;
+                if (!Array.isArray(content?.via)) return;
+                const parent = this.matrixClient.getRoom(ev.getStateKey());
+                if (canonicalOnly && !content?.canonical) return;
+                if (parent.currentState.maySendStateEvent(EventType.SpaceChild, userId)) {
```

Comment:
Comment: I find it pretty damn weird that the filter is "only show the linkage if you *could* have created this link now", but as noted in https://github.com/vector-im/element-web/issues/10935#issuecomment-898909964 that's what the spec says

Reply:
Given its weird, it might make a lot of sense to leave a comment either quoting or linking to the part of the spec that specifies the behaviour